marlowe-dashboard-client/src/Marlowe/Execution.purs

```diff
@@ -188,4 +211,4 @@ extractNamedActions state (When cases timeout cont)
 -- In reality other situations should never occur as contracts always reduce to When or Close
 -- however someone could in theory publish a contract that starts with another Contract constructor
 -- and we would want to enable moving forward with Evaluate
-extractNamedActions _ _ = [ Evaluate mempty ]
```

Comment:
I suggest to use `currentSlot` instead of `minSlot` above

Reply:
Currently, when you call "apply-inputs" endpoint in BE, Marlowe contract uses `SlotInterval currSlot (currSlot + 10)`.
I suggest to do the same, and use Pablo's suggestion.
```
env = makeEnvironment curSlot (curSlot + 10)
-- and instead of case computeTransaction emptyTx state cont
case reduceContractUntilQuiescent env state cont of

```

Reply:
Alternative would be

`env = makeEnvironment curSlot timeout`

Reply:
today we talked with @palas about this, we need to make the interval in `mkTx` and the `makeEnvironment` here match, and for that we will first call `Semantic.timeouts` on the contract to see when the `minTime` and `maxTime`, and if we have a `minTime` that is bigger than the currentSlot then do [currentSlot, minTime - 1], and if we don't then use [currentSlot, currentSlot + 10]